d/ec2_instance_type (#13124)
terraform-providers/terraform-provider-aws: website/docs/d/ec2_instance_type.html.markdown

---
subcategory: "EC2"
layout: "aws"
page_title: "AWS: aws_ec2_instance_type"
description: |-
  Information about single EC2 Instance Type.
---

# Data Source: aws_ec2_instance_type

> Use this data source to get characteristics about single EC2 Instance Type.

Reviewer: Nit: wording

```suggestion
Get characteristics for a single EC2 Instance Type.
```

## Example Usage

```hcl
data "aws_ec2_instance_type" "example" {
  instance_type = "t2.micro"
}

```

## Argument Reference

> The following arguments is supported:

Reviewer: ```suggestion
The following argument is supported:
```

* `instance_type` - (Required) Instance 

## Attribute Reference

> In addition to argument above, the following attributes are exported:

Reviewer: ```suggestion
In addition to the argument above, the following attributes are exported:
```

> ~> **NOTE:** Some values are not always set and may not be available for
> interpolation.

Reviewer: ```suggestion
~> **NOTE:** Not all attributes are set for every instance type.
```
